buildkite/pipelines/bacalhau-golang.yaml

```diff
@@ -0,0 +1,64 @@
+cache:
+  paths:
+    - "/golangci-lint-cache"
+    - "/vendor-cache"
+    - "/node-modules-cache"
+  size: 20g
+  name: "vendor-module-cache"
+
+steps:
+  - label: ":golang: Lint"
+    command: "./buildkite/scripts/lint.sh"
+    key: "lint"
+    agents:
+      queue: "buildkite-hosted-linux-medium"
+      
+  - group: ":package: Build Tarball"
+    key: "build-tarball"
+    steps:
+      - label: ":golang: Build linux amd64"
+        command: "./buildkite/scripts/build_tarball.sh linux amd64"
+        agents:
+          queue: "buildkite-hosted-linux-small"
+          
+      - label: ":golang: Build linux arm64"
+        command: "./buildkite/scripts/build_tarball.sh linux arm64"
+        agents:
+          queue: "buildkite-hosted-linux-small"
```

Comment:
how are steps in the same group executed? Do each require spinning up their own container, which means slower start and cost, or are they run within the same agent/container? I am double checking because we define the agents in each step, which implies they run on separate environments

Reply:
They are on different agents. Spinning up the agent is pretty fast. It stays in queue for max 5-6s. Group step only helps seeing things visually !